src/_Classics_/caeser_cipher/index.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Most simplest encryption scheme. Read more: [http://practicalcryptography.com/ciphers/caesar-cipher/]
+ * @param {String} str
+ * @param {Number} num
+ */
+
+function caeserCipher(str, num) {
+  const lowerCaseString = str.toLowerCase();
+  const alphabets = 'abcdefghijklmnopqrstuvwxyz'.split('');
+  const totalAlphabets = alphabets.length;
+  let result = '';
+
+  // handle large number, like 300 or -300
+  num %= totalAlphabets;
+
+  for (let index in lowerCaseString) {
+    // get the current character
+    const currentCharacter = lowerCaseString[index];
+
+    // if character is space, add it to the result and continue to next
+    if (currentCharacter === ' ') {
+      result += currentCharacter;
+      continue;
+    }
+
+    // determine the new index
+    const currentIndex = alphabets.indexOf(currentCharacter);
```

Comment:
Try to use Map instead of indexOf.
-  With indexOf complexity will be O(n*26).
-  With Map complexity will be O(n).

Reply:
@TheSTL, any example snippet would be helpful.

Reply:
![as](https://user-images.githubusercontent.com/50075905/64493004-67368900-d298-11e9-92fb-11f50f8e52a6.png)